### PR TITLE
fix(core): missing resource `meta` in route creation

### PR DESCRIPTION
**Pull request by aliemir**
When creating the routes with parameters internally, we were missing the `resource.meta` values which causes missing parameters when user defines a default parameter value in `resource.meta` and doesn't have any values in current `meta` or `params`.

### Self Check before Merge

Please check all items below before review.

-   [x] Corresponding issues are created/updated or not needed
-   [x] Docs are updated/provided or not needed
-   [x] Examples are updated/provided or not needed
-   [x] TypeScript definitions are updated/provided or not needed
-   [x] Tests are updated/provided or not needed
-   [x] Changesets are provided or not needed

### CODE REVIEW

**Review by alicanerdurmaz — APPROVED** [review]

File: packages/core/src/hooks/navigation/index.ts [hunk 1]
```diff
@@ -76,7 +84,15 @@ export const useNavigation = () => {
             }
 
             return go({
-                to: composeRoute(createActionRoute, parsed, meta),
+                to: composeRoute(
+                    createActionRoute,
+                    {},
```
Comment (BatuhanW) [inline packages/core/src/hooks/navigation/index.ts:89]: Why don't we pass parsed now?

File: packages/core/src/hooks/router/use-get-to-path/index.ts [hunk 2]
```diff
@@ -49,7 +49,15 @@ export const useGetToPath = (): GetToPathFn => {
                 return undefined;
             }
 
-            const composed = composeRoute(actionRoute, parsed, meta);
+            const composed = composeRoute(
+                actionRoute,
+                {},
```
Comment (BatuhanW) [inline packages/core/src/hooks/router/use-get-to-path/index.ts:54]: Seems like we always pass empty object here -at least in this PR-, should we remove this param if it's not needed?

**Comment by BatuhanW** [comment]
Would be nice to have some tests.

File: packages/core/src/hooks/router/use-get-to-path/index.ts (hunk 2 above)
Comment (aliemir) [reply packages/core/src/hooks/router/use-get-to-path/index.ts]: Updated with a change in `composeRoute` helper.

File: packages/core/src/definitions/helpers/router/compose-route.ts [hunk 3]
```diff
@@ -12,13 +11,25 @@ import { prepareRouteParams } from "./prepare-route-params";
...
     // prepareRouteParams (from route params, params and meta)
-    const preparedRouteParams = prepareRouteParams(routeParams, params, meta);
+    const preparedRouteParams = prepareRouteParams(routeParams, {
+        ...resourceMeta,
+        ...(typeof parsed?.id !== "undefined" ? { id: parsed?.id } : {}),
+        ...(typeof parsed?.action !== "undefined"
+            ? { action: parsed?.action }
+            : {}),
+        ...(typeof parsed?.resource !== "undefined"
+            ? { resource: parsed?.resource }
```
Comment (BatuhanW) [inline packages/core/src/definitions/helpers/router/compose-route.ts:28]:
```suggestion
            ? { resource: parsed.resource }
```

**Review by BatuhanW — APPROVED** [review]

**Pull request merged**